### PR TITLE
refactor(status-aggregator): unified federated type

### PR DESCRIPTION
**Pull request by JackZxj**

### CODE REVIEW

File: pkg/controllers/statusaggregator/controller.go [hunk 1]
```diff
@@ -20,192 +19,229 @@ package statusaggregator
...
-	// plugin for source type to aggregate statuses
-	plugin  plugins.Plugin
 	metrics stats.Metrics
 	logger  klog.Logger
+
+	dynamicClient dynamicclient.Interface
+	fedClient     fedclient.Interface
+
+	federatedInformer              informermanager.FederatedInformerManager
+	federatedObjectInformer        fedcorev1a1informers.FederatedObjectInformer
```
Comment (mrlihanbo) [inline pkg/controllers/statusaggregator/controller.go:81]: We name the informers as `fedObjectInformer` and `clusterFedObjectInformer` in other controllers, should we keep the consistent naming style?

File: pkg/controllers/statusaggregator/controller.go [hunk 2]
```diff
@@ -20,192 +19,229 @@ package statusaggregator
...
-		&util.ClusterLifecycleHandlerFuncs{
-			ClusterAvailable: func(cluster *fedcorev1a1.FederatedCluster) {
+
+	genericFederatedObjectHandler := eventhandlers.NewTriggerOnAllChanges(func(fedObj fedcorev1a1.GenericFederatedObject) {
+		if !fedObj.GetDeletionTimestamp().IsZero() {
+			return
+		}
+		unsObj, err := fedObj.GetSpec().GetTemplateAsUnstructured()
+		if err != nil {
+			a.logger.Error(err, "Failed to get object matadata")
```
Comment (mrlihanbo) [inline pkg/controllers/statusaggregator/controller.go:152]: typo

File: pkg/controllers/statusaggregator/controller.go [hunk 3]
```diff
@@ -20,192 +19,229 @@ package statusaggregator
...
+				}
+				a.worker.EnqueueWithDelay(reconcileKey{
+					gvk:       gvk,
+					namespace: obj.GetNamespace(),
+					name:      obj.GetName(),
+				}, a.objectEnqueueDelay)
+			})
+		},
+	}); err != nil {
+		return nil, fmt.Errorf("failed to create target object informer: %w", err)
```
Comment (mrlihanbo) [inline pkg/controllers/statusaggregator/controller.go:212]:
```suggestion
		return nil, fmt.Errorf("failed to add event handler generator: %w", err)
```

File: pkg/controllers/util/propagationstatus/propagationstatus.go [hunk 4]
```diff
@@ -27,12 +26,11 @@ import (
 	utiljson "k8s.io/apimachinery/pkg/util/json"
 	"k8s.io/apimachinery/pkg/util/sets"
 
-	fedtypesv1a1 "github.com/kubewharf/kubeadmiral/pkg/apis/types/v1alpha1"
+	"github.com/kubewharf/kubeadmiral/pkg/apis/core/v1alpha1"
```
Comment (limhawjia) [inline pkg/controllers/util/propagationstatus/propagationstatus.go:29]:
```suggestion
	fedcorev1a1 "github.com/kubewharf/kubeadmiral/pkg/apis/core/v1alpha1"
```

**Review by mrlihanbo — APPROVED** [review]

**Review by limhawjia — APPROVED** [review]

**Pull request merged**